### PR TITLE
Glow in the dark computer screens

### PR DESCRIPTION
**Pull request by Flaborized**
[FEATURE]

## what? 

This PR makes computer and computer3 type computers have screens that glow in the dark, using -overlays-. Did not do other computer-like faux-computer objects (like the chemaster) because there are a whole lot in all kinds of files. Someone else can do that in the future if they want 👁️ 

## why? 

Looks cool! matches how screens actually appear in the dark

## Changelog

```changelog
(u)Flaborized
(*)Most computer screens now glow in the dark.
```

### CODE REVIEW

File: code/modules/networks/computer3/computer3.dm [hunk 1]
```diff
@@ -33,6 +33,10 @@
 	var/setup_os_string = null
 	var/setup_font_color = "#19A319"
 	var/setup_bg_color = "#1B1E1B"
+
+	var/glow_in_dark_screen = 1 // does it have a glow in the dark screen? see computer_screens.dmi
```
Comment (ZeWaka) [inline code/modules/networks/computer3/computer3.dm:37]:
```suggestion
	var/glow_in_dark_screen = TRUE // does it have a glow in the dark screen? see computer_screens.dmi
```

File: code/obj/machinery/computer.dm [hunk 2]
```diff
@@ -8,6 +8,9 @@
 	var/light_r =1
 	var/light_g = 1
 	var/light_b = 1
+
+	var/glow_in_dark_screen = 1 // does it have a glow in the dark screen? see computer_screens.dmi
```
Comment (ZeWaka) [inline code/obj/machinery/computer.dm:12]:
```suggestion
	var/glow_in_dark_screen = TRUE // does it have a glow in the dark screen? see computer_screens.dmi
```

**Review by ZeWaka — CHANGES_REQUESTED** [review]
.

File: code/modules/networks/computer3/computer3.dm [hunk 3]
```diff
@@ -314,6 +318,12 @@
 
 		src.base_icon_state = src.icon_state
 
+		if(glow_in_dark_screen)
+			src.screen_image = image('icons/obj/computer_screens.dmi', src.icon_state, -1)
+			src.overlays += image(screen_image)
+			screen_image.plane = PLANE_LIGHTING
+			src.UpdateOverlays(screen_image, "screen_image")
```
Comment (ZeWaka) [inline code/modules/networks/computer3/computer3.dm:325]: ummmmmmm this doesn't look right!!!!

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/networks/computer3/computer3.dm (hunk 3 above)
Comment (ZeWaka) [reply code/modules/networks/computer3/computer3.dm:325]: you should not be using updateoverylays with manaul overlay adding and clearing

Comment (Flaborized) [reply code/modules/networks/computer3/computer3.dm]: I resolved this by removing updateoverlays, turns out that I didn't need them at all

**Comment by Gannets** [comment]
Looks fantastic, really good change.

File: code/modules/networks/computer3/computer3.dm (hunk 3 above)
Comment (ZeWaka) [reply code/modules/networks/computer3/computer3.dm]: i meant the opposite 😳

Comment (Flaborized) [reply code/modules/networks/computer3/computer3.dm]: ! i dont know what you mean then or how to fix

Comment (Sovexe) [reply code/modules/networks/computer3/computer3.dm]: one moment

File: code/modules/networks/computer3/computer3.dm [hunk 4]
```diff
@@ -33,6 +33,10 @@
 	var/setup_os_string = null
 	var/setup_font_color = "#19A319"
 	var/setup_bg_color = "#1B1E1B"
+
+	var/glow_in_dark_screen = TRUE // does it have a glow in the dark screen? see computer_screens.dmi
```
Comment (Sovexe) [inline code/modules/networks/computer3/computer3.dm:37]:
```suggestion
	/// does it have a glow in the dark screen? see computer_screens.dmi
	var/glow_in_dark_screen = TRUE
```

**Review by Sovexe — CHANGES_REQUESTED** [review]

File: code/obj/machinery/computer.dm [hunk 5]
```diff
@@ -45,6 +48,11 @@
 	light.set_color(light_r, light_g, light_b)
 	light.attach(src)
 
+	if(glow_in_dark_screen)
+		src.screen_image = image('icons/obj/computer_screens.dmi', src.icon_state, -1)
+		src.overlays += image(screen_image)
```
Comment (Sovexe) [inline code/obj/machinery/computer.dm:53]:
```suggestion
		src.UpdateOverlays(screen_image, "screen_image")
```

File: code/obj/machinery/computer.dm [hunk 6]
```diff
@@ -90,18 +98,26 @@
 		icon_state = initial(icon_state)
 		src.icon_state += "b"
 		light.disable()
+		if(glow_in_dark_screen)
+			src.overlays = null
```
Comment (Sovexe) [inline code/obj/machinery/computer.dm:102]:
```suggestion
			src.ClearSpecificOverlays("screen_image")
```

File: code/obj/machinery/computer.dm [hunk 7]
```diff
@@ -90,18 +98,26 @@
...
 
 	else if(powered())
 		icon_state = initial(icon_state)
 		status &= ~NOPOWER
 		light.enable()
+		if(glow_in_dark_screen)
+			src.overlays = null
+			src.overlays += image(screen_image)
```
Comment (Sovexe) [inline code/obj/machinery/computer.dm:110]:
```suggestion
			src.UpdateOverlays(screen_image, "screen_image")
```

File: code/obj/machinery/computer.dm [hunk 8]
```diff
@@ -90,18 +98,26 @@
...
+			screen_image.plane = PLANE_LIGHTING
 	else
 		SPAWN_DBG(rand(0, 15))
 			//src.icon_state = "c_unpowered"
 			icon_state = initial(icon_state)
 			src.icon_state += "0"
 			status |= NOPOWER
 			light.disable()
+			if(glow_in_dark_screen)
+				src.overlays = null
```
Comment (Sovexe) [inline code/obj/machinery/computer.dm:120]:
```suggestion
				src.ClearSpecificOverlays("screen_image")
```

**Review by Sovexe — CHANGES_REQUESTED** [review]
other half

File: code/obj/machinery/computer.dm [hunk 9]
```diff
@@ -8,6 +8,9 @@
 	var/light_r =1
 	var/light_g = 1
 	var/light_b = 1
+
+	var/glow_in_dark_screen = TRUE // does it have a glow in the dark screen? see computer_screens.dmi
```
Comment (Sovexe) [inline code/obj/machinery/computer.dm:12]:
```suggestion
	/// does it have a glow in the dark screen? see computer_screens.dmi
	var/glow_in_dark_screen = TRUE
```

**Review by Sovexe — APPROVED** [review]

**Comment by BatElite** [comment]
I quite like it but I do wonder if it'll mess with item ejected by the computers. Genetics would be the main thing and there's disks that some use.

**Pull request merged**